modules/http/scanner.go

```diff
@@ -161,6 +162,25 @@ func (scan *scan) withDeadlineContext(ctx context.Context) context.Context {
 func (scan *scan) dialContext(ctx context.Context, net string, addr string) (net.Conn, error) {
 	dialer := zgrab2.GetTimeoutConnectionDialer(scan.scanner.config.Timeout)
 
+	// If the scan is for a specific IP, and a domain name is provided, we
+	// don't want to just let the http library resolve the domain.  Create a
+	// fake resolver that we will use, that always returns the IP we are given
+	// to scan.
+	if scan.target.IP != nil && scan.target.Domain != "" {
+		// In the case of redirects, we don't want to blindly use the IP we
+		// were given to scan, however.  So only use the fake resolver if the
+		// domain originally specified for the scan target machines the current
+		// address being looked up in this DialContext.
+		targetAddr := fmt.Sprintf("%s:%d", scan.target.Domain, scan.getPort())
+		if addr == targetAddr {
```

Comment:
One exception to this exception -- what about a redirect to the same host, but a different port (80-443 for example)? It feels like it could just compare the host part of the addr, but I haven't gamed out what other weird cases that might open up.

Reply:
Resolved, see https://github.com/zmap/zgrab2/pull/240#discussion_r373658429